File: 729/main.go
package main

import (
	"fmt"
)



func main() {
	obj := Constructor()
	// fmt.Println(obj.Book(47, 50))
	// fmt.Println(obj.Book(33, 41))
	// fmt.Println(obj.Book(39, 45))
	// fmt.Println(obj.Book(33, 42))

	// fmt.Println(obj.Book(19, 25))
	// fmt.Println(obj.Book(18, 27))	// false

	fmt.Println(obj.Book(37, 50))
	fmt.Println(obj.Book(33, 50))
	fmt.Println(obj.Book(60, 70))
	fmt.Println(obj.Book(10, 20))
}

type MyCalendar struct {
    l [][]int
}


func Constructor() MyCalendar {
    return MyCalendar{l: [][]int{}}
}


func (this *MyCalendar) Book(startTime int, endTime int) bool {
    for _, v := range this.l {
		if endTime > v[0] && startTime < v[1] {
			return false
		}
	}
	this.l = append(this.l, []int{startTime, endTime})
	return true
}

File: 729/main.rs

struct MyCalendar {
    l:  Vec<Vec<i32>>,
}


/** 
 * `&self` means the method takes an immutable reference.
 * If you need a mutable reference, change it to `&mut self` instead.
 */
impl MyCalendar {

    fn new() -> Self {
        MyCalendar{l: vec![]}
    }
    
    fn book(&mut self, start_time: i32, end_time: i32) -> bool {
        for i in &self.l {
            if start_time < i[1] && end_time > i[0] {
                return false;
            }
        }
        let _ = &self.l.push(vec![start_time, end_time]);
        true
    }
}


fn main () {
    let mut obj = MyCalendar::new();
    let ret_1: bool = obj.book(10, 20);
    let ret_2: bool = obj.book(15, 20);
    let ret_3: bool = obj.book(30, 50);
    println!("{}", ret_1);
    println!("{}", ret_2);
    println!("{}", ret_3);
}